src/CalcManager/Ratpack/rat.cpp

```diff
@@ -54,9 +54,14 @@ void gcdrat( PRAT *pa, int32_t precision)
         }
 
     destroynum( pgcd );
-    *pa=a;
 
-    RENORMALIZE(*pa);
+    RENORMALIZE(a);
+
+    if (rat_equ(*pa, a, precision))
+    {
+        DUPRAT(*pa, a);
```

Comment:
This check works, but it seems like it's just a catch-all for errors caused by bad inputs. It would be preferable to do a better job of validating the inputs instead, as noted in the comment above.

If it isn't possible to just validate our inputs due to assumptions made elsewhere in code, let's have a conversation then about what the right fix will look like.